### ISSUE TITLE
Unicode support?

### ISSUE THREAD

**Issue by Cerzs** [nunuhara/xsystem4#44]
Hello there. 

Any plans to add unicode mode?

As I can see, Rance series is translatable into other languages, with the exception of 02 (II remake) and VI, as those do not have a .fnl file that can be modified. Or is there another way?

**Comment by nunuhara** [comment]
Unicode support is something I've always wanted to add "eventually" but I don't have any specific plans to work on it in the immediate future. I think it would be tricky to get it right, given the amount of libraries that deal with text and all of the external file formats used in system 4 games.

xsystem4 does support custom fonts (see the `--font-mincho` and `--font-gothic` options), so in theory you could make custom .ttf fonts that replace some Japanese characters with the characters that you need.

**Comment by Cerzs** [comment]
@nunuhara does it work on Windows?

**Comment by nunuhara** [comment]
>does it work on Windows?

It.. seems to? I hadn't tried building for Windows before now but I've [just released](https://github.com/nunuhara/xsystem4/releases/tag/alpha-411022b) a build which seems to work. I played up to the prison in Rance VI without issue, but I haven't tested any more than that.

>Edit: getting this error with Rance Vi

xsystem4 does not currently work with the MangaGamer version of VI or Sengoku Rance, due to some additional libraries that they used which haven't been implemented in xsystem4 yet (e.g. the "LoadCG" library mentioned in that error message). You would have to use the original Japanese version as a base.

**Comment by Cerzs** [comment]
@nunuhara I see. I'll try then. I was testing your windows build and seems compatible with 02 kai eng version for now. The font seems a bit... broken tho lol. Maybe I should try the japanese version too.
![imagen](https://user-images.githubusercontent.com/44489935/160303677-674bb2ae-88ff-455c-9657-bca523cb751a.png)

**Comment by nunuhara** [comment]
Hmm that's odd. They must have changed something in the text rendering code for the MangaGamer version. The old fan-translated version of 02 and the original Japanese version don't have this issue.

Also I think there's some bugs in the Windows build relating to the encoding of save file names, so saving probably won't work on some games if the save file names include Japanese characters. I won't have time to fix this right away but I'll try to do it within the next week or so.

**Comment by Cerzs** [comment]
Thank you. I'll test some more. If I find something that looks like a bug, I'll post here

**Comment by Cerzs** [comment]
I can say R02 works "fine". As you said, savefile is broken. Also I got this... Idk what it is, a png but the background turns black? (Sill's attack) 
![imagen](https://user-images.githubusercontent.com/44489935/160914598-0ab3d12c-7370-4f67-bbd9-ac76df88a4d1.png)

I think it also happen with others effects, but I don't know if this is a "bug".

**Comment by nunuhara** [comment]
That was indeed a bug.
I've fixed a number of bugs in [the latest release](https://github.com/nunuhara/xsystem4/releases/tag/alpha-1), including this transparency bug, the text layout bug in the English version, and issues with saving/loading.

Note however that the fix for text layout issues in the English version causes text to flow outside of the text box sometimes.

**Comment by Cerzs** [comment]
Thank you! I will test it. 

Btw, It's just an idea, about the text overflow, maybe a temp solution would be adding -fontsize? Example: `--font-gothic -fontsize 24` 

I don't know how doable it is, it just occurred to me.

**Comment by nunuhara** [comment]
The `-fontsize` idea doesn't really work because xsystem4 has no way of distinguishing between which text goes in the message box vs which text is for the UI, etc.

Instead, I've added a `--font-x-scale` option which will scale all text horizontally. I've found `--font-x-scale=0.9375` works pretty well for Rance 02, so I've set that as the default for that game.
The English version of Rance VI is also working now in the [latest release](https://github.com/nunuhara/xsystem4/releases/tag/alpha-2).
